### PR TITLE
feat: add selections to binned data

### PR DESCRIPTION
**Pull request by jjxie**
Add tap, lasso and range selections for binned data.
Need this https://github.com/qlik-trial/qlik-chart-modules/pull/87 to make it work

https://user-images.githubusercontent.com/25456307/142165474-7a87999d-e244-4c47-b66e-f35532713dca.mov

### CODE REVIEW

**Comment by jjxie** [comment]
This is ready for review

File: src/models/tick-model/ticks/tick-helper.js [hunk 1]
```diff
@@ -42,9 +42,12 @@ const tickHelper = {
   },
 
   getSize(dockService, chartModel, chart, dimension) {
+    const dataHandler = chartModel.query.getDataHandler();
     const size = chartModel.query.isPrelayout()
       ? dockService.meta.chart.size[dimension]
-      : chart.component(KEYS.COMPONENT.POINT).rect[dimension];
+      : chart.component(dataHandler.getMeta().isBinnedData ? KEYS.COMPONENT.HEAT_MAP : KEYS.COMPONENT.POINT).rect[
+          dimension
+        ];
```
Comment (veinfors) [inline src/models/tick-model/ticks/tick-helper.js:50]: should not be line breaks here... should look like:
`....YS.COMPONENT.POINT).rect[dimension];`

Comment (jjxie) [reply src/models/tick-model/ticks/tick-helper.js:50]: Seems Eslint did it automatically

File: src/utils/bins-util.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+const populateBins = (dataPages, bins) => {
+  // eslint-disable-next-line no-param-reassign
+  bins.length = 0;
+  const matrix = dataPages[0].qMatrix;
+  let i;
+
+  // Hack for snapshot
```
Comment (veinfors) [inline src/utils/bins-util.js:7]:
```suggestion
```

File: src/utils/bins-util.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  if (dataPages[0].reformatted) {
+    for (i = 1; i < matrix.length; i++) {
+      try {
+        bins.push(matrix[i][0]);
+      } catch (err) {
+        // console.log( err );
```
Comment (veinfors) [inline src/utils/bins-util.js:13]:
```suggestion
```

File: src/utils/bins-util.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      }
+    }
+  } else {
+    for (i = 1; i < matrix.length; i++) {
+      try {
+        matrix[i][0].qText = JSON.parse(matrix[i][0].qText);
+        bins.push(matrix[i][0]);
+      } catch (err) {
+        // console.log( err );
```
Comment (veinfors) [inline src/utils/bins-util.js:22]:
```suggestion
```

**Review by veinfors — APPROVED** [review]

**Pull request merged**